packages/x-components/src/composables/__tests__/use-debounce.spec.ts

```diff
@@ -0,0 +1,73 @@
+import { mount } from '@vue/test-utils';
+import { DebounceOptions } from '../../utils/types';
+import { useDebounce } from '../use-debounce';
+
+const fnMock = jest.fn();
+const fnParamStub = 'It is a parameter';
+
+function render(debounceTimeInMs = 200, debounceOptions: DebounceOptions = {}) {
+  const wrapper = mount({
+    setup: () => {
+      const debouncedFnMock = useDebounce(
+        (param: string) => fnMock(param),
+        debounceTimeInMs,
+        debounceOptions
+      );
+      const onClick = () => debouncedFnMock(fnParamStub);
+      return { onClick };
+    },
+    template: `<button @click="onClick">Execute fn debounced</button>`
+  });
+
+  return {
+    wrapper,
+    runDebouncedFnMock: async () => await wrapper.trigger('click')
+  };
+}
+
+describe('testing useDebounce composable', () => {
+  jest.useFakeTimers();
+
+  beforeEach(() => {
+    fnMock.mockClear();
+  });
+
+  it('should debounce the fn with the debounced time', async () => {
+    const { runDebouncedFnMock } = render();
+
+    await runDebouncedFnMock();
+    expect(fnMock).toHaveBeenCalledTimes(0);
+    jest.advanceTimersByTime(200);
+    expect(fnMock).toHaveBeenCalledTimes(1);
+    expect(fnMock).toHaveBeenCalledWith(fnParamStub);
+
+    fnMock.mockClear();
+    await runDebouncedFnMock();
+    jest.advanceTimersByTime(100);
+    await runDebouncedFnMock();
+    jest.advanceTimersByTime(150);
+    expect(fnMock).toHaveBeenCalledTimes(0);
+    jest.advanceTimersByTime(50);
+    expect(fnMock).toHaveBeenCalledTimes(1);
```

Comment:
I'd test again `expect(fnMock).toHaveBeenCalledWith(fnParamStub);`

Reply:
Done and aligned with main